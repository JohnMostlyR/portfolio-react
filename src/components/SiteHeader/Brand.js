import React from 'react';
import styled from 'styled-components';
import Media from '../MediaObject/Media';
import typography from '../../helpers/styles/templates/typography';

import MyLogo from '../../images/my-logo.svg';

const BrandWrapper = styled.div`
  float: left;
  padding: 1vh 0 0 1vw;
`;

const BrandLink = styled.a.attrs({
  href: props => props.href || '#',
})`
  display: inline-block;
  width: 45px;
  color: #fff;
  text-decoration: none;
  
  ${typography.minion}
`;

const Brand = (props) => (
  <BrandWrapper>
    <BrandLink href={'/intro'}>
      <Media
        bodyAlign='middle'
        imageSource={MyLogo}
        imageAlt='logo'
        imageAlign='middle'
        imageHeight='45px'
        imageWidth='45px'
      >JOHAN MEESTER</Media>
    </BrandLink>
  </BrandWrapper>
);

export default Brand;
