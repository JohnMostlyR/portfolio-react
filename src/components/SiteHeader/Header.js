import styled from 'styled-components';

const Header = styled.header`
  width: 100%;
  z-index: 10;
  background-color: #575756;
`;

export default Header;