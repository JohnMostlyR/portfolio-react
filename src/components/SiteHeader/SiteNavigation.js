import React, {Component} from 'react';
import {NavLink} from 'react-router-dom';
import {throttle} from 'lodash';
import 'scrolling-element';
import styled from 'styled-components';
import mq from '../../helpers/styles/templates/mediaQueries';
import typography from '../../helpers/styles/templates/typography';

const SiteNavigationNav = styled.nav`
  padding: 1vh 0;
  background-color: #575756;

  ${mq.m`
    left: 100%;
    position: absolute;
    top: 77px;
    transform: rotate(90deg);
    transform-origin: top left;
    background-color: transparent;
    border-bottom: none;  
  `}
`;

const SiteNavigationUL = styled.ul`
  display: flex;
  justify-content: space-around;
  list-style: none;
  margin: 0;
  padding: 0;

  ${mq.m`justify-content: flex-start;`}
`;

const SiteNavigationLI = styled.li`
  vertical-align: middle;

  ${mq.m`margin-top: 2vw;`}

  &:not(:first-child) {
    ${mq.m`margin-left: 1vh;`}
  }
`;

const SiteNavigationNavLink = styled(NavLink)`
  display: inline-block;
  position: relative;
  background-color: #FF6633;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  padding: 0.5em;
  text-align: center;
  text-decoration: none !important;
  transition: background-color 0.3s, color 0.3s;

  ${typography.pica}
  
  &:active,
  &:hover {
    background-color: transparent;
    color: #FF6633;

    &::before {
      opacity: 1;
      transform: scale3d(1, 1, 1);
    }
  }

  &::before {
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: 1;
    border: 1px solid #FF6633;
    border-radius: inherit;
    opacity: 0;
    transform: scale3d(0.6, 0.6, 1);
    transition: transform 0.3s, opacity 0.3s;
    transition-timing-function: cubic-bezier(0.75, 0, 0.125, 1);

    ${mq.m`
      border: 2px solid #FF6633;
      border-radius: 5px;
    `}
  }

  ${mq.m`
    border-radius: 5px;
    width: 7rem; // For IE and Edge who don't support the max-content prop.
    width: max-content;
  `}
  
  &.is-active {
    background-color: transparent;
    color: #FF6633;

    &::before {
      opacity: 1;
      transform: scale3d(1, 1, 1);
    }
  }
`;

const SiteNavigationIcon = styled.span`
  display: none !important;

  ${mq.m`
    display: inherit !important;
    transform: rotate(-90deg);
    margin: 0;
  `}
`;

const SiteNavigationDescription = styled.span`
  ${mq.m`margin-left: .5rem;`}

  @media all and (orientation: landscape) and (min-width: 37.5em) {
    display: none;
  }

  @media all and (orientation: landscape) and (min-height: 37.5em) {
    display: inline;
  }
`;

class SiteNavigation extends Component {
  state = {
    myTopPosition: 0,
    topPositionOffset: 0,
  };

  _getElementTop = (element) => {
    let actualTop = element.offsetTop;
    let currentParent = element.offsetParent;

    while (currentParent !== null) {
      actualTop += currentParent.offsetTop;
      currentParent = currentParent.offsetParent;
    }

    return actualTop;
  };

  handleScrollEvent = (ev) => {
    let delta = (this.state.myTopPosition - ev.target.scrollingElement.scrollTop);
    if (delta <= 0) {
      this.setState({topPositionOffset: Math.abs(delta)});
    } else {
      this.setState({topPositionOffset: 0});
    }
  };

  componentDidMount() {
    const myTopPosition = this._getElementTop(this.siteNavigation);
    this.setState({myTopPosition});
    document.addEventListener('scroll', throttle(this.handleScrollEvent, 2));
  }

  componentWillUnmount() {
    document.removeEventListener('scroll', this.handleScrollEvent);
  }

  render() {
    const pages = [
      {
        description: 'Intro',
        icon: 'fa fa-home fa-fw',
        navigateTo: 'intro',
      },
      {
        description: 'About',
        icon: 'fa fa-address-card fa-fw',
        navigateTo: 'about',
      },
      {
        description: 'Skills',
        icon: 'fa fa-code fa-fw',
        navigateTo: 'skills',
      },
      {
        description: 'Projects',
        icon: 'fa fa-list fa-fw',
        navigateTo: 'projects',
      },
      {
        description: 'Contact',
        icon: 'fa fa-send fa-fw',
        navigateTo: 'contact',
      },
    ];

    const setStyle = () => {
      if (this.state.topPositionOffset) {
        return (
          {
            position: 'absolute',
            top: (this.state.myTopPosition + this.state.topPositionOffset + 16),
            zIndex: 10,
          }
        );
      } else {
        return ({});
      }
    };

    return (
      <SiteNavigationNav style={setStyle()} innerRef={el => this.siteNavigation = el}>
        <SiteNavigationUL>
          {
            pages.map((page, idx) => {
              return (
                <SiteNavigationLI key={idx}>
                  <SiteNavigationNavLink
                    to={`/${page.navigateTo}`}
                    activeClassName='is-active'
                  ><SiteNavigationIcon
                    className={`${page.icon} ${this.props.className}`}
                    aria-hidden="true"/><span><SiteNavigationDescription>{page.description}</SiteNavigationDescription></span>
                  </SiteNavigationNavLink>
                </SiteNavigationLI>
              );
            })
          }
        </SiteNavigationUL>
      </SiteNavigationNav>
    );
  }
}

export default SiteNavigation;
