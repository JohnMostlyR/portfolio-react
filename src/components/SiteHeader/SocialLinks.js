import React, {Component} from 'react';
import styled from 'styled-components';
import typography from '../../styles/templates/typography';
import {visuallyHidden} from '../../styles/mixins';

const SocialLinksAside = styled.aside`
  float: right;
  padding: 0 1em 0 4em;
  background-image: linear-gradient(45deg, transparent 23%, #fff 23%);
  color: #575756;
`;

const SocialLinksH2 = styled.h2`${visuallyHidden}`;

const SocialLinksList = styled.ul`
  display: flex;
  list-style: none;
`;

const SocialLinksListItem = styled.li`
  padding: .5em;
  transition: transform .2s;

  &:hover {
    transform: scale(1.5);
  }

  ${typography.doublePica}
  
  @media (min-width: 37.5em) and (max-height: 31.25em) {
    ${typography.pica}
  }
`;

const SocialLinksLink = styled.a`
  color: #575756;
  text-decoration: none;
`;

const VisuallyHiddenSpan = styled.span`
  ${visuallyHidden}
`;

class SocialLinks extends Component {
  playAnimation = (links) => {
    const INDEX = 0; // starting node index
    const START_DELAY = 3000; // milliseconds;
    const DELAY = 200; // milliseconds;

    setTimeout(function a(nodes, idx) {
      nodes[idx].style.transform = 'scale(1.5)';

      setTimeout(() => {
        nodes[idx].style.removeProperty('transform');
        if (idx < nodes.length - 1) {
          setTimeout(a, DELAY, nodes, idx += 1);
        } else {
          return true;
        }
      }, DELAY);
    }, START_DELAY, links, INDEX);
  };

  componentDidMount() {
    if (this.list.children.length) {
      this.playAnimation(this.list.children);
    }
  }

  render() {
    const externalLinks = [
      {
        icon: 'fa fa-free-code-camp',
        name: 'freeCodeCamp',
        url: 'https://www.freecodecamp.org/mensae',
      },
      {
        icon: 'fa fa-github',
        name: 'Github',
        url: 'https://github.com/Mensae',
      },
      {
        icon: 'fa fa-linkedin-square',
        name: 'LinkedIn',
        url: 'https://nl.linkedin.com/in/jmeester',
      },
      {
        icon: 'fa fa-codepen',
        name: 'Codepen',
        url: 'https://codepen.io/jmeester',
      },
    ];

    return (
      <SocialLinksAside>
        <SocialLinksH2>Find me on:</SocialLinksH2>
        <SocialLinksList innerRef={e => this.list = e}>
          {
            externalLinks.map((externalLink, idx) => (
              <SocialLinksListItem key={idx} data-tip={externalLink.name}>
                <SocialLinksLink href={externalLink.url} target="_blank"
                   rel="noopener noreferrer">
                  <span className={externalLink.icon} aria-hidden="true"/><VisuallyHiddenSpan>{externalLink.name}</VisuallyHiddenSpan>
                </SocialLinksLink>
              </SocialLinksListItem>
            ))
          }
        </SocialLinksList>
      </SocialLinksAside>
    );
  }
}

export default SocialLinks;
