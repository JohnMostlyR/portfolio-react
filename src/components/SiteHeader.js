import React from 'react';
import SiteNavigation from '../components/SiteNavigation';
import SocialLinks from '../components/SocialLinks';

import MyLogo from '../images/my-logo.svg';

const SiteHeader = (props) => {
  return (
    <header role="banner" className="c-portfolio-global-header">
      <div className="c-portfolio-page-header__banner u-clearfix">
        <div className="c-portfolio-global-header__brand">
          <a className="c-portfolio-global-header__link" href={'/intro'}>
            <div className='o-media'>
              <img className="o-media__img c-portfolio-global-header__brand-logo" src={MyLogo} alt="logo"/>
              <div className='o-media__body'>
                <span>JOHAN</span><br/><span>MEESTER</span>
              </div>
            </div>
          </a>
        </div>
        <SocialLinks/>
      </div>
      <SiteNavigation/>
    </header>
  );
};

export default SiteHeader;
