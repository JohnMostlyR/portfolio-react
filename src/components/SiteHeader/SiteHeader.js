import React from 'react';
import styled from 'styled-components';
import {clearfix} from '../../helpers/styles/mixins';
import Brand from './Brand';
import SiteNavigation from './SiteNavigation';
import SocialLinks from './SocialLinks';

const StyledHeader = styled.header`
  width: 100%;
  z-index: 10;
  background-color: #575756;
`;

const Wrapper = styled.div`
  border-bottom: .5vh solid #575756;
  
  ${clearfix}
`;

const SiteHeader = (props) => {
  return (
    <StyledHeader role="banner">
      <Wrapper>
        <Brand/>
        <SocialLinks/>
      </Wrapper>
      <SiteNavigation/>
    </StyledHeader>
  );
};

export default SiteHeader;
