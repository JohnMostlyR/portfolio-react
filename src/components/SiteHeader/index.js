import React from 'react';
import PropTypes from 'prop-types';

import Header from './Header';
import Wrapper from './Wrapper';

import Brand from '../Brand';
import SiteNavigation from '../SiteNavigationBar';
import SocialLinks from '../SocialLinks';

const SiteHeader = ({setSiteNavIsFixedOffset, siteNavIsAtScreenTop}) => (
    <Header role="banner">
      <Wrapper>
        <Brand/>
        <SocialLinks/>
      </Wrapper>
      <SiteNavigation setSiteNavIsFixedOffset={setSiteNavIsFixedOffset} siteNavIsAtScreenTop={siteNavIsAtScreenTop}/>
    </Header>
);

SiteHeader.propTypes = {
  setSiteNavIsFixedOffset: PropTypes.func.isRequired,
  siteNavIsAtScreenTop: PropTypes.bool,
};

export default SiteHeader;
