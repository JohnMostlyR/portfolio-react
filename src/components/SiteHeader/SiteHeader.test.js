import React from 'react';
import 'jest-styled-components';
import {shallow} from 'enzyme';
import createRouterContext from 'react-router-test-context';

import SiteHeader from './index';

describe('SiteHeader', () => {
  const context = createRouterContext();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
        <SiteHeader setSiteNavIsFixedOffset={(offset) => offset}/>,
        {context}
    );
  });


  it('Should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});