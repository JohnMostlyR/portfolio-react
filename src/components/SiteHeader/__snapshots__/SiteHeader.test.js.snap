// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader Should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SiteHeader
    setSiteNavIsFixedOffset={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <Brand
                    href="/"
                    imageSource="my-logo.svg"
                    text="JOHAN MEESTER"
          />
          <Unknown />
</styled.div>,
        <Unknown
          setSiteNavIsFixedOffset={[Function]}
/>,
      ],
      "role": "banner",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Brand
              href="/"
              imageSource="my-logo.svg"
              text="JOHAN MEESTER"
/>,
            <Unknown />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "href": "/",
              "imageSource": "my-logo.svg",
              "text": "JOHAN MEESTER",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "setSiteNavIsFixedOffset": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <Brand
                        href="/"
                        imageSource="my-logo.svg"
                        text="JOHAN MEESTER"
            />
            <Unknown />
</styled.div>,
          <Unknown
            setSiteNavIsFixedOffset={[Function]}
/>,
        ],
        "role": "banner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Brand
                href="/"
                imageSource="my-logo.svg"
                text="JOHAN MEESTER"
/>,
              <Unknown />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "href": "/",
                "imageSource": "my-logo.svg",
                "text": "JOHAN MEESTER",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "setSiteNavIsFixedOffset": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "_listeners": Array [],
          "_notifyListeners": [Function],
          "action": "POP",
          "createHref": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "dfixqz",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {
            "hash": "",
            "key": "dfixqz",
            "pathname": "/",
            "search": "",
          },
          "match": Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          },
        },
        "staticContext": undefined,
      },
    },
  },
}
`;
